packages/aws-rfdk/lib/deadline/lib/repository.ts

```diff
@@ -501,6 +535,11 @@ export class Repository extends Construct implements IRepository {
    */
   private readonly installerGroup: AutoScalingGroup;
 
+  /**
+   * Deadline Secrets Management settings.
+   */
+  public readonly secretsManagementSettings: SecretsManagementProps
```

Comment:
I don't think there's any use case where Secrets Management is enabled but the credentials are unset, or the credentials are set but Secrets Management is disabled, so we can simplify this to be an optional field that holds the credentials and we can infer if SM is enabled based on if it's set.

Reply:
I can imagine when user want to set Secrets Management is enabled (just to make sure) but expect secret is created by RFDK. Anyway such structure looks clear for understanding.

Reply:
We should definitely have it on the SecretsManagementProps interface, but storing the whole interface on the class instance rather than just the credentials isn't necessary. You can leave the interface as-is but change this to just store the `credentials` optional.

If I want to tell if Secrets Management is enabled I could just check if `credentials` is set rather than first checking the value of `enabled` and then checking the value of `credentials`. Also, in this context, enabled isn't optional, it must be set, but on the interface it's optional, so it isn't really what we need here.

Reply:
Actually it is a valid case for enabled to be true and the credentials to be undefined, we will generate them and save them in Secrets Manager in that case. The only case that wouldn't be defined is if enabled was `false` but credentials were supplied. I don't really see that case as being a problem, the expected behaviour there would be that SM is disabled and the credentials are ignored.

Reply:
For my opinion it is much clear and understandable to have separate parameters to enable SM and credentials.
Also this interface can be extended by others parameters in the future.

Reply:
My personal preference would be two interfaces, one for the construct properties and one for the class variable, but they would be so close to the same thing that I feel it's a bit nit-picky and not worth the effort. I'd say just leave the interface and the class variable as-is.

For reference, I was thinking we would have these interfaces:
```
export interface SecretsManagementProps {
  readonly enabled?: boolean;
  readonly credentials?: ISecret;
}
export interface SecretsManagementSettings {
  readonly enabled: boolean;
  readonly credentials?: ISecret;
}
```
then their usage would be:
```
export interface RepositoryProps {
  readonly secretsManagementSettings?: SecretsManagementProps;
  ...
}

export class Repository extends Construct implements IRepository {
  public readonly secretsManagementSettings: SecretsManagementSettings;

  constructor(scope: Construct, id: string, props: RepositoryProps) {
      this.secretsManagementSettings = {
      enabled: props.secretsManagementSettings?.enabled ?? true,
      credentials: props.secretsManagementSettings?.credentials ??
        ((props.secretsManagementSettings?.enabled ?? true) ? new Secret(this, 'SMAdminUser', {
          description: 'Admin credentials for Secret Management',
          generateSecretString: {
            excludeCharacters: '\"$&\'()/<>[\\]\`{|}',
            includeSpace: false,
            passwordLength: 24,
            requireEachIncludedType: true,

            generateStringKey: 'password',
            secretStringTemplate: JSON.stringify({ username: Repository.DEFAULT_SECRETS_MANAGEMENT_USERNAME }),
          },
        }) : undefined),
    };
    ...
  }

As you can see, the only difference is that `enabled` isn't optional on the class variable.

Reply:
Hm. Actually I can't see any reason to make `enabled` property optional.
If someone decided to define `secretsManagementSettings` what the reason do not set `enabled`?
What do you say if I just make this property mandatory for both cases?